src/postgres_client/postgres_client_account_index.rs

```diff
@@ -0,0 +1,419 @@
+            }
+
+            measure.stop();
+            inc_new_counter_debug!(
+                "accountsdb-plugin-postgres-update-mint-index-us",
+                measure.as_us() as usize,
+                10000,
+                10000
+            );
+            inc_new_counter_debug!(
+                "accountsdb-plugin-postgres-update-mint-index-count",
+                self.batch_size,
+                10000,
+                10000
+            );
+        }
+        Ok(())
+    }
+
+    /// Generic function to queue the token owner index for bulk insert.
+    fn queue_token_owner_index_gen<G: GenericTokenAccount>(
+        &mut self,
+        token_id: &Pubkey,
+        account: &DbAccountInfo,
+    ) {
+        if account.owner() == token_id.to_bytes() {
+            if let Some(owner_key) = G::unpack_account_owner(account.data()) {
+                let owner_key = owner_key.to_bytes().to_vec();
+                let pubkey = account.pubkey();
+                self.pending_token_owner_index.push(TokenSecondaryIndex {
+                    owner: owner_key,
+                    inner_key: pubkey.to_vec(),
+                    slot: account.slot,
+                });
+            }
+        }
+    }
+
+    /// Generic function to queue the token mint index for bulk insert.
+    fn queue_token_mint_index_gen<G: GenericTokenAccount>(
+        &mut self,
+        token_id: &Pubkey,
+        account: &DbAccountInfo,
+    ) {
+        if account.owner() == token_id.to_bytes() {
+            if let Some(mint_key) = G::unpack_account_mint(account.data()) {
+                let mint_key = mint_key.to_bytes().to_vec();
+                let pubkey = account.pubkey();
+                self.pending_token_mint_index.push(TokenSecondaryIndex {
+                    owner: mint_key,
+                    inner_key: pubkey.to_vec(),
+                    slot: account.slot,
+                })
+            }
+        }
+    }
+
+    /// Queue bulk insert secondary indexes: token owner and token mint indexes.
+    pub fn queue_secondary_indexes(&mut self, account: &DbAccountInfo) {
+        if self.index_token_owner {
+            self.queue_token_owner_index_gen::<inline_spl_token::Account>(
+                &inline_spl_token::id(),
+                account,
+            );
+            self.queue_token_owner_index_gen::<inline_spl_token_2022::Account>(
+                &inline_spl_token_2022::id(),
+                account,
+            );
+        }
+
+        if self.index_token_mint {
+            self.queue_token_mint_index_gen::<inline_spl_token::Account>(
+                &inline_spl_token::id(),
+                account,
+            );
+            self.queue_token_mint_index_gen::<inline_spl_token_2022::Account>(
+                &inline_spl_token_2022::id(),
+                account,
+            );
+        }
+    }
+
+    /// Generic function to update a single token owner index.
+    fn update_token_owner_index_gen<G: GenericTokenAccount>(
```

Comment:
nit: not too sure what the `_gen` suffix stands for, seems like we can just remove it

Reply:
done